### PR TITLE
Added Websocket's broadcast queue load metrics

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

Now we send metrics when the websocket's broadcast queue receive or
consume a message from the queue

Screenshot from 2020-02-24 18-06-45

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-22046

### CODE REVIEW

File: app/web_hub.go [hunk 1]
```diff
@@ -433,6 +436,9 @@ func (h *Hub) Start() {
 					}
 				}
 			case msg := <-h.broadcast:
+				if metrics := h.app.Metrics(); metrics != nil {
+					metrics.DecrementWebSocketBroadcastHubElements(strconv.Itoa(h.connectionIndex), 1)
```
Comment (ethervoid) [inline app/web_hub.go]: I'm not sure if we should use a goroutine to call these functions. The prometheus client should do it asynchronously but this one is inside a select:

```go
h.app.Srv().Go(func() {
    metrics.DecrementWebSocketBroadcastHubElements(strconv.Itoa(h.connectionIndex), 1)
})
```

Comment (agnivade) [reply app/web_hub.go]: There's no need to use a goroutine. The increment/decrement happens in-memory. The numbers are later scraped from the /metrics endpoint.

File: einterfaces/metrics.go [hunk 2]
```diff
@@ -37,6 +37,8 @@ type MetricsInterface interface {
 
 	IncrementWebsocketEvent(eventType string)
 	IncrementWebSocketBroadcast(eventType string)
+	IncrementWebSocketBroadcastHubElements(hub string, amount float64)
```
Comment (agnivade) [inline einterfaces/metrics.go]: May be (Increment|Decrement)WebSocketBroadcastBufferSize is a better name ?

File: app/web_hub.go (hunk 1 above)
Comment (ethervoid) [reply app/web_hub.go]: So [in this case](https://github.com/mattermost/mattermost-server/blob/master/app/web_conn.go#L235-L237) is not needed

Comment (agnivade) [reply app/web_hub.go]: Whoa boy .. yes indeed. Let's remove that. Would you be open to send a PR ?

Comment (ethervoid) [reply app/web_hub.go]: Sure! That's why I was confused about using goroutines

**Pull request merged**